### PR TITLE
fix(runs): retry provider streams that complete without any output (AYC-678)

### PR DESCRIPTION
**Pull request by pavellnk**
- A stream that finishes cleanly with zero chunks (empty provider
  response) previously failed the run outright as RUN_EXECUTION_FAILED
  'No final message received from streaming inference' (incident #548)
- An empty completed attempt showed and persisted nothing, so it now
  gets the same one-shot retry as a stall (AYC-652) or a pre-output
  token-limit truncation (AYC-669)
- A second empty attempt still falls through to the orchestrator's
  classified RUN_EXECUTION_FAILED, so the incident signal survives
- Extract pure chunk-accumulation logic into
  stream-accumulation.helper.ts to keep the service under the 500-line
  file cap

Refs: AYC-645, AYC-674

---

> [!NOTE]
> **Medium Risk**
> Changes core run streaming/retry behavior for empty provider responses; limited to one retry when nothing was yielded or persisted, with new tests covering edge cases.
> 
> **Overview**
> **Empty provider streams** that finish without emitting any chunks no longer fail the run immediately. `StreamingInferenceService` tracks `yieldedContent` on each attempt; if the first attempt completes without yielding anything (nothing shown or persisted), it logs a warning and performs **one** retry—same safety model as stall/token-limit retries before durable output. A second empty attempt still exits with no yields so the orchestrator can surface `RUN_EXECUTION_FAILED` (incident #548).
> 
> Chunk folding (text/thinking/tool-call deltas, `AccumulatedState`) is moved into **`stream-accumulation.helper.ts`** so the service stays under the file size cap; behavior is unchanged aside from setting `yieldedContent` when a chunk triggers a UI update.
> 
> Specs cover: retry then success on empty→text, cap at two execute calls when both attempts are empty, and no retry when text streamed normally.
> 
> <sup>Reviewed by [Cursor Bugbot](https://cursor.com/bugbot) for commit b559417a1c7641dee4d1e6b909ea76b7165650f5. Bugbot is set up for automated code reviews on this repo. Configure [here](https://www.cursor.com/dashboard/bugbot).</sup>

### CODE REVIEW

**Comment by pavellnk** [comment]
* **#1270** <a href="https://app.graphite.com/github/pr/ayunis-core/ayunis-core/1270?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈 <a href="https://app.graphite.com/github/pr/ayunis-core/ayunis-core/1270?utm_source=stack-comment-view-in-graphite" target="_blank">(View in Graphite)</a>
* **#1258** <a href="https://app.graphite.com/github/pr/ayunis-core/ayunis-core/1258?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#1257** <a href="https://app.graphite.com/github/pr/ayunis-core/ayunis-core/1257?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#1256** <a href="https://app.graphite.com/github/pr/ayunis-core/ayunis-core/1256?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* `main`


<h2></h2>

This stack of pull requests is managed by <a href="https://graphite.dev?utm-source=stack-comment"><b>Graphite</b></a>. Learn more about <a href="https://stacking.dev/?utm_source=stack-comment">stacking</a>.

File: ayunis-core-backend/src/domain/runs/application/services/streaming-inference.service.ts [hunk 1]
```diff
@@ -137,11 +126,7 @@ export class StreamingInferenceService {
         },
       );
     }
-    yield* this.streamAttempt(
-      params,
-      { producedOutput: false },
-      assistantMessage,
-    );
+    yield* this.streamAttempt(params, freshTracker(), assistantMessage);
```
[the remark this answers is not in the document]
Comment (pavellnk) [reply ayunis-core-backend/src/domain/runs/application/services/streaming-inference.service.ts:129]: Intentional, not double-counting: `messageId` is a log-only tag in `CollectUsageAsyncService.collect()` — it is not part of `TokensConsumedEvent` and the usage record schema has no messageId column or unique constraint. Usage rows are append-only and each attempt's row reflects tokens the provider actually billed (an empty completed response still consumes input tokens), so recording both attempts is correct cost accounting — the same per-attempt pattern the merged token-limit retry (#1257) uses. Dropping the first attempt's usage would under-report real provider spend.
